test/fixtures/SpokePool.Fixture.ts

```diff
@@ -297,17 +305,29 @@ export async function modifyRelayHelper(
   depositId: string,
   originChainId: string,
   depositor: SignerWithAddress
-): Promise<{ messageHash: string; signature: string }> {
-  const messageHash = ethers.utils.keccak256(
-    defaultAbiCoder.encode(
-      ["string", "uint64", "uint32", "uint32"],
-      ["ACROSS-V2-FEE-2.0", modifiedRelayerFeePct, depositId, originChainId]
-    )
-  );
-  const signature = await depositor.signMessage(ethers.utils.arrayify(messageHash));
-
+): Promise<{ signature: string }> {
+  const typedData = {
+    types: {
+      UpdateRelayerFeeMessage: [
+        { name: "newRelayerFeePct", type: "uint64" },
+        { name: "depositId", type: "uint32" },
+        { name: "originChainId", type: "uint256" },
+      ],
+    },
+    domain: {
+      name: "ACROSS-V2",
+      version: "1.0.0",
+      chainId: Number(originChainId),
+      verifyingContract: "0xCcCCccccCCCCcCCCCCCcCcCccCcCCCcCcccccccC",
+    },
+    message: {
+      newRelayerFeePct: modifiedRelayerFeePct,
+      depositId,
+      originChainId,
+    },
+  };
+  const signature = await depositor._signTypedData(typedData.domain, typedData.types, typedData.message);
```

Comment:
1. This is how we'd sign the speed up in the FE after this change goes through? Is there a non underscore function we can use instead?
2. Do we need any additional unit tests? Like verifying that the typed data has the expected fields and can be parsed by a wallet client?

Reply:
> 1. This is how we'd sign the speed up in the FE after this change goes through? Is there a non underscore function we can use instead?

Yes, we would use this in the FE to speed up. This is just a wrapper method provided by [ethers](https://docs.ethers.org/v5/api/signer/#Signer-signTypedData) and will eventually be replaced by `signTypedData` without an underscore. Alternatively, we could just directly use the JSON RPC like:
```js
const signature = await depositor.provider.send("eth_signTypedData_v4", [fromAddress, encodedMsgParams]
```
where we need to do some additional encoding. This approach is a bit verbose though.

> 2. Do we need any additional unit tests? Like verifying that the typed data has the expected fields and can be parsed by a wallet client?

Not sure if I understand correctly. Verifying whether the typed data has the expected fields is already covered in our tests, no? Basically, the test case is where if the signed message is not equal to the expected message then the verification fails. I am also not sure how to write a unit test on whether or not a wallet client can parse the typed data. Doesn't this depend on the implementation of the wallet? And can't we assume that if a wallet supports the method `eth_signTypedData_v4` then it can parse any message supported by the spec?

Reply:
> eventually be replaced by signTypedData without an underscore

Any idea when this is going to happen? Not that it blocks us i'm just curious what's holding this up

Reply:
> Verifying whether the typed data has the expected fields is already covered in our tests, no? Basically, the test case is where if the signed message is not equal to the expected message then the verification fails

Do we need to perform additional checks that the signed message is in the correct EIP712 format?

Reply:
> > eventually be replaced by signTypedData without an underscore
> 
> Any idea when this is going to happen? Not that it blocks us i'm just curious what's holding this up

No idea tbh... Couldn't find anything on this in the repo.



> > Verifying whether the typed data has the expected fields is already covered in our tests, no? Basically, the test case is where if the signed message is not equal to the expected message then the verification fails
> 
> Do we need to perform additional checks that the signed message is in the correct EIP712 format?

Is this possible? The signature is in the end just the same bytes string as a message signed with `signer.signMessage`.